simple_rpc/simple_rpc.py

```diff
@@ -182,6 +182,7 @@ def __init__(self, device, baudrate=9600):
         """
         super().__init__(device, baudrate)
         self.open()
+        self._close()
```

Comment:
I don't get it. The `open` function is wrapped by `_auto_open`, so a double close should not be needed, right?

Reply:
Sorry this was a bit hard to see. I removed the `SocketInterface` open because I assumed it had some kind of recursive problem given the traceback message I got. I didn't look at it very closely though to see what the cause was.

Reply:
Ah, I see. Both `open` and `_get_methods` are now wrapped and `open` calls `_get_methods`. Removing the method override for `_get_methods` should do the trick.

Can you test the latest version?

Reply:
Sorry for the late reply, have been out on vacation the past few days. The latest version seems to work well, thanks!

Reply:
I did not expect anything before the new year. Thanks for testing.

I added a little wrapper for backwards compatibility and to make the CLI work again. I will add some documentation later.